### PR TITLE
Fix padding for int contexts

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* On Linux, the final call to `.to` creates trouble when input tensors are integer. For example:

```
>>> a = torch.tensor([1])
>>> b = torch.stack([torch.full((1,), torch.nan), a])
>>> b
tensor([[nan],
        [1.]])
>>> b.to(a)
tensor([[-9223372036854775808],
        [                   1]])
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/test_utils.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    ],
+)
+def test_pad_and_stack(tensors: list):
+    stacked_and_padded = left_pad_and_stack_1D(tensors)
+    assert stacked_and_padded.shape == (len(tensors), max(len(t) for t in tensors))
+
+    ref = torch.concat(tensors).to(dtype=stacked_and_padded.dtype)
+
+    assert torch.sum(torch.nan_to_num(stacked_and_padded, nan=0)) == torch.sum(ref)
+    assert torch.nanmean(stacked_and_padded) == torch.nanmean(ref)
```
Comment (abdulfatir) [inline test/test_utils.py:24]: Why two assertions that seem to be checking more or less the same thing?

Comment (lostella) [reply test/test_utils.py:24]: I think the first one is more robust, because it will run also for int tensors

File: test/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    [
+        list(map(torch.tensor, [[1, 2, 3], [5, 6]])),
+        list(map(torch.tensor, [[2.0, 3.0], [4.0, 5.0, 6.0], [7.0, 8.0, 9.0, 10.0]])),
+    ],
+)
+def test_pad_and_stack(tensors: list):
+    stacked_and_padded = left_pad_and_stack_1D(tensors)
+    assert stacked_and_padded.shape == (len(tensors), max(len(t) for t in tensors))
+
+    ref = torch.concat(tensors).to(dtype=stacked_and_padded.dtype)
```
[the remark this answers is not in the document]
Comment (lostella) [reply test/test_utils.py:21]: can also do that

**Review by abdulfatir — APPROVED** [review]
